### PR TITLE
Add Configurations for SRL Application

### PR DESCRIPTION
**Pull request by bhargav**
- [x] Commit the training configuration files into saul-examples.
- [x] Read overloaded configuration from CLI arguments.

### CODE REVIEW

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SemanticRoleLabeling/SRLApps.scala [hunk 1]
```diff
@@ -47,25 +52,30 @@ object SRLApps extends Logging {
...
-      if (trainArgType && useGoldBoundaries && useGoldPredicate && trainingMode.equals("other")) "aTr"
+  val experimentName: String = {
+    if (trainingMode.equals("other")) {
+      if (trainArgType && useGoldBoundaries && useGoldPredicate) "aTr"
       else if (trainArgIdentifier && useGoldPredicate && useGoldPredicate) "bTr"
       else if (trainArgType && useGoldPredicate && !useGoldBoundaries) "cTr"
       else if (trainPredicates && useGoldPredicate) "dTr"
       else if (trainArgIdentifier && !useGoldPredicate) "eTr"
       else if (trainArgType && !useGoldPredicate) "fTr"
       else ""
```
Comment (danyaljj) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SemanticRoleLabeling/SRLApps.scala]: Might be unnecessary because your `require(experimentName.nonEmpty, "Experiment is not configured properly.")` down below....

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SemanticRoleLabeling/SRLApps.scala [hunk 2]
```diff
@@ -74,55 +84,66 @@ object SRLApps extends Logging {
...
+    require(
+      (trainPredicates ^ trainArgIdentifier ^ trainArgType) & !(trainPredicates & trainArgIdentifier & trainArgType),
+      "Training modes should be mutually exclusive."
+    )
+
+    experimentName match {
 
       case "aTr" =>
-        argumentTypeLearner.modelDir = modelDir + expName
+        argumentTypeLearner.modelDir = modelDir + experimentName
```
Comment (danyaljj) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SemanticRoleLabeling/SRLApps.scala]: love this renaming!

**Comment by danyaljj** [comment]
I like the overall changes; but sth that bothers me is the redundancy in the config files. Like all the config files have the addresses to WSJ data etc (which are not really part of classifier configuration).  Why don't we move them into a common configuration?

**Comment by kordjamshidi** [comment]
I think @christos-c made the current configuration to avoid these redundancies.

**Comment by bhargav** [comment]
I need separate files to define the parameter for each experiment. The common parts are: `Path to treebank, path to propbank, test section, train section[from, to]`. I can make the values from these files the default in the `SRLConfigurator` file but that would break the unit tests and people outside CogComp.

**Comment by danyaljj** [comment]
All look good. Maybe we create the common config for this PR?

**Comment by danyaljj** [comment]
Had a conversation with @bhargav; I am convinced that the current setting is simple enough and it's fine to keep it as it is. So it's all good by me to merge. 

@kordjamshidi if you don't have any objections, it's good to go.

**Comment by kordjamshidi** [comment]
The problem comes from the place that we want to have one App for each application. Given this idea we made an effort to remove the redundancies and have a compact configuration so made it a bit nesting to place all possible configurations in one app. Now, it seems you find it hard to follow, and went for another alternative to flatten all the settings and and repeat each flag value explicitly in each configuration as @bhargav did and hide an independent configuration file for each learning configuration!  It is ok to merge for me but , according to the way that I originally thought about Saul programs, it looks a bit odd to me. Because, I was thinking, each Saul program is ideally just a specification of a configuration. But we combined all configurations and now for each Saul running case  we have a separate configuration behind it. This was my objection when I was discussing with @christos-c about having one App for the whole SRL.  This is a bit philosophical but I don't mind merging this.

**Comment by kordjamshidi** [comment]
Please see here my [Scala configurator object here](https://github.com/IllinoisCogComp/saul/blob/9cf1f25e9d51ebdae34f4e25086c09485ccc5bb2/saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SemanticRoleLabeling/SRLApps.scala#L17-L67), if this is good enough then we can close this issue. Now it is an object inside App file but I can move it out.  @bhargav @danyaljj

**Comment by bhargav** [comment]
The Configurator object looks fine. But it does not simplify running each of the `aTr`, `bTr` etc. Ideal solution would be to run the configuration without having to change all corresponding parameters also. We can maybe have instantiations of the Configuration for each of the special running scenarios.

**Comment by kordjamshidi** [comment]
yes, right, I removed the redundancies sticking with the current structure. My goal was to remove one step of the resource manager. There were a lot of redundant variables. For me changing and experimenting with the current setting is very convenient . I think the description of each model is in the documentation. So if the "if then else" structure is confusing for the reader she/he can see the documentation. There is a table that says which flags are set for each model.

**Comment by bhargav** [comment]
Closing this PR as obsolete.
